shared/logger.go

```diff
@@ -143,14 +134,35 @@ func newAppEngineFlexContext(r *http.Request, project string, commonResource *mr
 // commonResource is an optional override to the monitored resource details appended to each log.
 // e.g. in the Flex environment, it pays to override this value to type gae_app, to ensure finding
 // logs is consistent between services.
-func HandleWithGoogleCloudLogging(h http.HandlerFunc, project string, commonResource *mrpb.MonitoredResource) http.HandlerFunc {
+func HandleWithGoogleCloudLogging(h http.HandlerFunc, gcClient *gclog.Client, project string, childrenLogger, parentLogger *gclog.Logger) http.HandlerFunc {
 	return func(w http.ResponseWriter, r *http.Request) {
-		ctx, err := newAppEngineFlexContext(r, project, commonResource)
+		start := time.Now()
+		// See https://cloud.google.com/appengine/docs/flexible/go/writing-application-logs
+		traceID := strings.Split(r.Header.Get("X-Cloud-Trace-Context"), "/")[0]
+		if traceID != "" {
+			traceID = fmt.Sprintf("projects/%s/traces/%s", project, traceID)
+		}
+		ctx, err := newAppEngineFlexContext(r, gcClient, traceID, childrenLogger)
 		if err != nil {
 			h(w, r)
 			return
 		}
 		h(w, r.WithContext(ctx))
+
+		end := time.Now()
+		e := gclog.Entry{
+			Timestamp: end,
+			Trace:     traceID,
+			Severity:  gclog.Info,
+			HTTPRequest: &gclog.HTTPRequest{
+				Request: r,
+				Status:  0,
```

Comment:
```suggestion
				// TODO: set Status based on w
```
0 is the default. Let's not set it again and leave a comment instead.